packages/accounts-base/accounts_server.js

```diff
@@ -77,9 +77,10 @@ export class AccountsServer extends AccountsCommon {
     // user expects. The way to make this work in a publish is to do
     // Meteor.find(this.userId).observe and recompute when the user
     // record changes.
-    var currentInvocation = DDP._CurrentInvocation.get();
+    var currentInvocation = DDP._CurrentPublicationInvocation.get();
+    currentInvocation = currentInvocation || DDP._CurrentInvocation.get();
     if (!currentInvocation)
-      throw new Error("Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.");
+      throw new Error("Meteor.userId can only be invoked in method calls or publications.");
```

Comment:
Yes. I think the argument here is that publications are longer active, so there is more chance that content returned by `user` changes.

BTW, I have a [package which gives server-side reactivity inside publish functions](https://github.com/peerlibrary/meteor-reactive-publish), so then `Meteor.user()` is reactive. :-)